frontend/src/metabase-types/store/state.ts

```diff
@@ -16,7 +16,22 @@ import type { SettingsState } from "./settings";
 import type { SetupState } from "./setup";
 import type { FileUploadState } from "./upload";
 
-type modalName = null | "collection" | "dashboard" | "action";
+type ModalName = null | "collection" | "dashboard" | "action";
+
+// TODO: convert redux/undo and UndoListing.jsx to TS and update type
+type UndoState = {
```

Comment:
Please move to `undo.ts`

Reply:
[`278dd4c` (#43884)](https://github.com/metabase/metabase/pull/43884/commits/278dd4c3cd7aa91f910d2800f4b63f9797da847b)